### PR TITLE
Create Test Vectors

### LINKED ISSUE #36: Test tbdex-kt/tbdex-js interop
Create a set if tbDEX test vectors to add to the tbdex repo. Then test protocol compatibility between JS and Kotlin

### PR DESCRIPTION
**Pull request by tomdaffurn**
with @mistermoe 
Closes #36 
- Create test vectors JSON file
- Synchronize tbDEX json-schema with tbdex-js
- Use Networknt instead of Everit for JSON schema validation
- Remove nulls from serialised tbDEX messages

### CODE REVIEW

**Review by mistermoe — APPROVED** [review]
💯 🏅 💰

File: protocol/src/main/resources/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (jiyoontbd) [inline protocol/src/main/resources/index.html:1]: where is this used?

**Review by jiyoontbd — APPROVED** [review]

File: protocol/src/main/kotlin/tbdex/sdk/protocol/Json.kt [hunk 2]
```diff
@@ -1,61 +1,14 @@
...
-import com.fasterxml.jackson.databind.JsonNode
-import com.fasterxml.jackson.databind.JsonSerializer
+import com.fasterxml.jackson.annotation.JsonInclude
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.databind.ObjectWriter
 import com.fasterxml.jackson.databind.SerializationFeature
-import com.fasterxml.jackson.databind.SerializerProvider
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
-import typeid.TypeID
+import tbdex.sdk.protocol.serialisation.TypeIdModule
```
Comment (phoebe-lew) [inline protocol/src/main/kotlin/tbdex/sdk/protocol/Json.kt:8]: this spelling of serialization 🇦🇺

File: protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt [hunk 3]
```diff
@@ -46,13 +48,14 @@ class Rfq private constructor(
...
+    matchedOfferingMethod.requiredPaymentDetails?.let {
+      val schema = matchedOfferingMethod.getRequiredPaymentDetailsSchema()
+      val jsonNodePaymentDetails = Json.jsonMapper.valueToTree<JsonNode>(selectedMethod.paymentDetails)
+      schema?.validate(jsonNodePaymentDetails)
     }
-
   }
 
-  private fun verifyClaims(requiredClaims: List<PresentationDefinitionV2>) {
+  private fun verifyClaims(requiredClaims: PresentationDefinitionV2) {
```
Comment (phoebe-lew) [inline protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt:58]: if multiple VCs required, are we smushing into a single PD?

File: protocol/src/main/kotlin/tbdex/sdk/protocol/serialisation/TypeIdModule.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import com.fasterxml.jackson.databind.Module
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.module.SimpleModule
+import typeid.TypeID
+import java.io.IOException
+
+/**
+ * Jackson [Module] the serialise [TypeID] fields.
+ */
+class TypeIdModule: SimpleModule() {
```
Comment (phoebe-lew) [inline protocol/src/main/kotlin/tbdex/sdk/protocol/serialisation/TypeIdModule.kt:18]: nice

File: protocol/src/test/kotlin/tbdex/sdk/protocol/models/OfferingTest.kt [hunk 5]
```diff
@@ -47,6 +49,8 @@ class OfferingTest {
     val offering = TestData.getOffering()
     offering.sign(TestData.ALICE_DID)
 
+    println(Json.jsonMapper.convertValue<JsonNode>(offering).toPrettyString())
```
Comment (phoebe-lew) [inline protocol/src/test/kotlin/tbdex/sdk/protocol/models/OfferingTest.kt:52]: debug?

**Review by phoebe-lew — APPROVED** [review]

File: protocol/src/main/resources/index.html (hunk 1 above)
Comment (tomdaffurn) [reply protocol/src/main/resources/index.html:1]: https://tbdex.dev/

File: protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt (hunk 3 above)
Comment (tomdaffurn) [reply protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt:58]: This change is to match the json schema. Maybe @mistermoe knows?

**Pull request merged**